### PR TITLE
feat(mod): make remote dynamic

### PR DESCRIPTION
**Pull request by harry-hov**
# Description

Add `remote` flag to `gnodev mod download` command. 
Default: `test3.gno.land:36657`

Related #462

### CODE REVIEW

File: cmd/gnodev/mod.go [hunk 1]
```diff
@@ -12,38 +12,62 @@ import (
...
+}
 
-	switch args[0] {
-	case "download":
-		if err := runModDownload(); err != nil {
-			return fmt.Errorf("mod download: %w", err)
-		}
-	default:
-		return fmt.Errorf("invalid command: %s", args[0])
+func execModDownload(cfg *modDownloadCfg, args []string, io *commands.IO) error {
```
Comment (zivkovicmilos) [inline cmd/gnodev/mod.go:66]: Why did you pass in the IO if you do not use it anywhere yet?

File: cmd/gnodev/mod.go [hunk 2]
```diff
@@ -12,38 +12,62 @@ import (
...
 
-func newModCmd() *commands.Command {
-	return commands.NewCommand(
+type modDownloadCfg struct {
+	remote string
+}
+
+var defaultModDownloadOptions = &modDownloadCfg{
+	remote: "test3.gno.land:36657",
+}
```
Comment (zivkovicmilos) [inline cmd/gnodev/mod.go:21]: If you don't share this configuration between files in this package, you can safely drop it and just have the default option be set when you are registering flags in `modDownloadCfg.RegisterFlags()`

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

I'd probably update the PR description to include just a bit more details on why we need the configurable `remote` for mod downloads, otherwise it's good to go 🚀

File: cmd/gnodev/mod.go (hunk 2 above)
Comment (harry-hov) [reply cmd/gnodev/mod.go]: Done 356b1fe8e2bfcfc5b484f7f2edd845919312a55a

File: cmd/gnodev/mod.go (hunk 1 above)
Comment (harry-hov) [reply cmd/gnodev/mod.go:62]: I was planning to add `verbose` flag in this PR. But some more work needs to be done before adding that. So dropped the idea. Can we keep it for future?

**Review by moul — APPROVED** [review]

**Pull request merged**